rpc/src/rpc.rs

```diff
@@ -5123,6 +5117,15 @@ pub mod tests {
         let request = create_test_request("getClusterNodes", None);
         let result: Value = parse_success_result(rpc.handle_request_sync(request));
         let expected = json!([{
+            "pubkey": rpc.identity.to_string(),
+            "gossip": "127.0.0.1:8000",
+            "shredVersion": 0u16,
+            "tpu": "127.0.0.1:8003",
+            "rpc": format!("127.0.0.1:{}", rpc_port::DEFAULT_RPC_PORT),
+            "pubsub": format!("127.0.0.1:{}", rpc_port::DEFAULT_RPC_PUBSUB_PORT),
+            "version": null,
+            "featureSet": null,
+        }, {
```

Comment:
Apparently previously `getClusterNodes` was filtering out the first result because the gossip address was not valid:
https://github.com/solana-labs/solana/blob/5d85a3459/rpc/src/rpc.rs#L3459

In order to get some other tests which need a valid `tvu` address to pass, I had to update L4689 to use `ContactInfo::new_localhost` instead of `ContactInfo::default`. The former creates valid address for all sockets whereas the latter does not. This has caused expected result in this test to change.